nidn/fdtd_integration/compute_spectrum_fdtd.py

```diff
@@ -172,26 +172,56 @@ def _average_along_detector(signal):
     return avg
 
 
-def _summed_thickness_times_permittivity(thicknesses, permittivity):
+def _summed_thickness_times_sqrt_permittivity(thicknesses, permittivity):
+    """
+    Helper function to calculate the sum of the product of the thickness and
+    the square root of the permittivity for each layer in a material stack.
+
+    Args:
+        thicknesses (tensor): tensor with the thickness of each layer of the material stack
+        permittivity (tensor): tensor with the real relative permittivity for each layer of the material stack
+
+    Returns:
+        float: sum of thickness times sqrt(e_r) for each layer
+    """
     summed_permittivity = 0
     for i in range(len(thicknesses)):
-        summed_permittivity += thicknesses[i] * permittivity[:, :, i].real
+        summed_permittivity += thicknesses[i] * torch.sqrt(permittivity.real.max())
     return summed_permittivity
 
 
-def _check_sufficient_timesteps(
-    cfg: DotMap, thicknesses, wavelength, permittivity, scaling
-):
-    recommended_timesteps = int(((
-        cfg.FDTD_free_space_distance
-        + 3 * _summed_thickness_times_permittivity(thicknesses, permittivity)
-        + 2 * wavelength * 1e6
-    ) / (1 / torch.sqrt(torch.tensor(2.0)) * (1 / scaling))).item()// 1)
+def _check_if_enough_timesteps(cfg: DotMap, wavelength, permittivity):
+    """
+    Function to find the recommended minimum number of timesteps.
+    The signal should have passed trough the material, been reflected to the start of the material and reflected again to the
+    rear detector before the signal is assumed to be steady state. The max permittivity is used for all layers, in case of patterned layers in the future.
+
+    Args:
+        cfg (DotMap): config
+        wavelength (float): wavelength of the simulation
+        permittivity (tensor): tensor of relative permittivities for each layer
+    """
+    wavelengths_of_steady_state_signal = 5
+    number_of_internal_reflections = 3
```

Comment:
Also comment here, this is an assumption I guess? Would it be different for more layers, thicker layers e.g.?